### ISSUE TITLE
Can I use bundletool.jar to test ANE-PlayAssetDelivery

### ISSUE THREAD

**Issue by zengqingyong** [airsdk/Adobe-Runtime-Support#2704]
Is my path wrong? Or ANE-PlayAssetDelivery can not be used for local testing, you must upload to google play to test, When I use bundletool jar installation var file: AssetFile = asset. OpenInstallTimeAsset (" assetpack1/0. Pak "); file is null
![5](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/8c348ca5-7e83-4116-9ecd-48f3c76c20c9)
![6](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/6cbc7748-6630-4701-9538-7e1a7e52af76)
![7](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/a2ee4a0e-0073-4148-a6f0-3e607d09fe98)
![8](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/4b69d833-85b9-4228-b080-0b91ccebe85e)

**Comment by ajwfrost** [comment]
From the looks of this, the path you should be using is `OpenInstallTimeAsset("0.Pak");`

The command in the app descriptor file is to take the `assetpack1` folder for this asset pack, which means all the contents of that folder will be packaged into the asset pack. So none of the file paths will start with that path name.

You can actually check this using a zip manager I believe, rather than needing the bundle tool. If you open up the AAB file using a zip manager, you will see the internal folder structure and can see the asset packs there and how the files are organised in that..

thanks

**Comment by zengqingyong** [comment]
![1](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/9ccfae12-da5a-4def-8730-0a6f69685e8a)
![2](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/47e32e5b-b4ff-4fbf-8eff-732b71fcf510)
Is this right？

**Comment by ajwfrost** [comment]
Looks like it's right, I assume within the "assets" folder there you have the "0.pak" files? but no further "assetpack1" folder?

I think that "assetpack1" at the top level of the bundle is given that name because of the asset pack `id` that you gave it in the app descriptor file. You can have a play around with that value and the folder name etc to see how the files actually end up getting put together..

thanks

**Comment by zengqingyong** [comment]
Yes, there is "0.pak" in the "assets" folder, but in my Logcat it looks like this
![22](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/df501f49-6e5a-460a-8f93-6ac0ca6faa02)
I just called 
asset = new PlayAssetDelivery();
			var boo:Boolean = asset.initAssetDelivery();
			if (boo)
			{
				txt0.text = "True";
			}else
			{
				txt0.text = "Flase";
			}
			trace("init>>>>>>>>>>>>>>>" + boo );

the  result is  true,but    var file:AssetFile = asset.openInstallTimeAsset("test0.swf");   the  result  is null

Do I need more ANE files?

**Comment by ajwfrost** [comment]
Hmm... can't see much wrong with that. Is it possible to get your AAB file and then we can check it here and look at the structures/replicate the set-up for this and debug...

thanks

**Comment by zengqingyong** [comment]
[test.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/11888798/test.zip)
[test01.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/11888827/test01.zip)
[test02.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/11888849/test02.zip)
Please rename like this 
![11](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/29392ed7-c62f-4a57-acf3-fdf1401f5b04)
Thank you!

**Comment by ajwfrost** [comment]
Thanks - so there is an asset pack called "assetpack1", but (a) this is on-demand rather than install-time, and (b) this only contains a single SWF file (which is 8MB in size).

So for that, you'd need to trigger the download first (`fetchAssetPack`), and then when that's available you can request the asset file path via `getAssetAbsolutePath` using the filename "test0.swf"

thanks

**Comment by zengqingyong** [comment]
Yes, when I uploaded aab to GooglePlay for testing, I used logcet to catch errors, as shown here
![1](https://github.com/airsdk/Adobe-Runtime-Support/assets/39934711/c44db7dc-6107-49f8-92dc-a8ed351d5a83)
var aa:int =  asset.fetchAssetPack("test0.swf"); the result is 0
